spec/classes/init_spec.rb

```diff
@@ -3285,6 +3285,63 @@
       end
     end
   end
+  
+  describe 'with ssh_key_export parameter specified' do
+    context 'as a non-boolean or non-string' do
+    let(:params) { { :ssh_key_export => ['not_a_boolean','or_a_string'] } }
+
+      it 'should fail' do
+        expect {
+          should contain_class('ssh')
+        }.to raise_error(Puppet::Error)
+      end
+    end
+
+    context 'as an invalid string' do
+      let(:params) { { :ssh_key_export => 'invalid_string' } }
+      let(:facts) do
+        { :osfamily          => 'RedHat',
+          :lsbmajdistrelease => '6',
+        }
+      end
+
+      it 'should fail' do
+        expect {
+          should contain_class('ssh')
+        }.to raise_error(Puppet::Error,/ssh::ssh_key_export may be either 'true' or 'false' and is set to <invalid_string>\./)
+      end
+    end
+
+    ['true',true].each do |value|
+      context "as #{value}" do
+        let(:params) { { :ssh_key_export => value } }
+        let(:facts) do
+          { :osfamily          => 'RedHat',
+            :lsbmajdistrelease => '6',
+          }
+        end
+
+        it { should compile.with_all_deps }
```

Comment:
Since ssh_key_import defaults to true, the sshkey that you export should be in the catalog. Could you try adding a resource for the sshkey for here with all the values, including the new host_aliases?